src/coreclr/inc/sospriv.idl

```diff
@@ -519,3 +519,13 @@ interface ISOSDacInterface14 : IUnknown
     HRESULT GetThreadStaticBaseAddress(CLRDATA_ADDRESS methodTable, CLRDATA_ADDRESS thread, CLRDATA_ADDRESS *nonGCStaticsAddress, CLRDATA_ADDRESS *GCStaticsAddress);
     HRESULT GetMethodTableInitializationFlags(CLRDATA_ADDRESS methodTable, MethodTableInitializationFlags *initializationStatus);
 }
+
+[
+    object,
+    local,
+    uuid(7ed81261-52a9-4a23-a358-c3313dea30a8)
+]
+interface ISOSDacInterface15 : IUnknown
+{
+    HRESULT GetMethodTableSlotMethodDesc(CLRDATA_ADDRESS mt, unsigned int slot, CLRDATA_ADDRESS *value);
```

Comment:
Sharing comment from offline conversation:

We may want to make the new API return (optional) code pointer, (optional) MethodDesc, and optional (Module,methodDef) pair. We should always have at least one of those. The caller can then figure out the details based on what's available.

This would make the API work in case we materialize MethodDescs lazily.